### PR TITLE
Private forums are now working via AP

### PR DESCRIPTION
**Pull request by annando**
Forum posts are now always transported via ActivityPub. Locally it should work without problems. It should also mostly work with older systems.

Further tests surely will be needed.

### CODE REVIEW

File: database.sql [hunk 1]
```diff
@@ -1612,7 +1612,6 @@ CREATE VIEW `post-user-view` AS SELECT
 	`post-user`.`deleted` AS `deleted`,
 	`post-user`.`origin` AS `origin`,
 	`post-thread-user`.`origin` AS `parent-origin`,
-	`post-thread-user`.`forum_mode` AS `forum_mode`,
```
Comment (MrPetovan) [inline database.sql:1615]: Shouldn't we bump the database version for this?

File: src/Model/Item.php [hunk 2]
```diff
@@ -2499,10 +2541,10 @@ public static function performActivity(int $item_id, string $verb, int $uid)
...
 			'verb'          => $activity,
 			'object-type'   => $objtype,
-			'allow_cid'     => $item['allow_cid'],
-			'allow_gid'     => $item['allow_gid'],
-			'deny_cid'      => $item['deny_cid'],
-			'deny_gid'      => $item['deny_gid'],
+			'allow_cid'     => $allow_cid,
+			'allow_gid'     => $allow_gid,
+			'deny_cid'      => $deny_cid,
+			'deny_gid'      => $deny_gid,
```
Comment (MrPetovan) [inline src/Model/Item.php]:
```suggestion
			'allow_cid'     => $allow_cid ?? $item['allow_cid'],
			'allow_gid'     => $allow_gid ?? $item['allow_gid'],
			'deny_cid'      => $deny_cid ?? $item['deny_cid'],
			'deny_gid'      => $deny_gid ?? $item['deny_gid'],
```

File: src/Model/Item.php [hunk 3]
```diff
@@ -2479,6 +2516,11 @@ public static function performActivity(int $item_id, string $verb, int $uid)
 			return true;
 		}
 
+		$allow_cid = $allow_cid ?? $item['allow_cid'];
+		$allow_gid = $allow_gid ?? $item['allow_gid'];
+		$deny_cid  = $deny_cid ?? $item['deny_cid'];
+		$deny_gid  = $deny_gid ?? $item['deny_gid'];
+
```
Comment (MrPetovan) [inline src/Model/Item.php]:
```suggestion
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: database.sql (hunk 1 above)
Comment (annando) [reply database.sql:1615]: We can. But it is not mandatory. But yeah, possibly I should.

**Comment by annando** [comment]
BTW: This is some breaking change. From now on the communities will only work with AP only. Because of this I added the checks for Diaspora. In some follow up PR I can also remove the community code from the DFRN part. In the next days I will perform several communication tests with the private communities. Locally they now work. And remotely they *should*. But for enhanced tests I need some more systems to test with.

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
Should I wait to merge this then?

**Comment by annando** [comment]
No. When I ask about the usage of private communities most answers had been "I have something local here". This I have tested.

**Pull request merged**